Libraries/CustomComponents/Navigator/Navigator.js

```diff
@@ -1043,6 +1043,14 @@ var Navigator = React.createClass({
     return this.state.routeStack.slice();
   },
 
+  getCurrentRouteIndex: function() {
+    return this.state.presentedIndex;
+  },
+
+  getCurrentRoute: function() {
+    return {...this.state.routeStack[this.getCurrentRouteIndex()]};
```

Comment:
Do we have to make a copy of a scene here? When you get an array of routes, we just make sure you do not mutate stack manually (however underlaying scenes are still the same).